src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+
+class Url implements Arrayable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public $scheme = null,
+        public $host = null,
+        public $port = null,
+        public $user = null,
+        public $pass = null,
+        public $path = null,
+        public $query = null,
+        public $fragment = null,
+    ) {
+    }
+
+    /**
+     * Parse a URL string into a URL object.
+     *
+     * @param  string  $url
+     * @return static
+     */
+    public static function parse($url)
```

Comment:
return types as well

Reply:
The framework doesn't currently use return types anywhere.

Reply:
The framework does not add return types on existing code as it would be a breaking change, but AFAIK there is no no-type policy

https://github.com/search?q=repo%3Alaravel%2Fframework+%3A+string&type=code